CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+# Contributing to this repository
+
+# Table of Contents
+1. [Commits and pull requests](#commits-release-notes-pull-requests)
+    1. [Fixing a bug](#fixing-a-bug)
+    2. [Commits](#commits)
+    3. [Pull requests](#pull-requests)
+        1. [Reviewing pull requests](#reviewing-pull-requests)
+        2. [Merging pull requests](#merging-pull-requests)
+
+## Commits and pull requests<a name="commits-release-notes-pull-requests"></a>
+### Fixing a bug<a name="fixing-a-bug"></a>
+
+1. Fork the repo and commit changes to a new branch in your fork.
+2. Make pull requests against the master branch in the rpc-gating repo.
+3. Reference the JIRA issue in both the commit message and the pull request message.
+
+### Commits<a name="commits"></a>
+
+Please avoid the following in a commit:
+
+* Mixing whitespace changes with functional code changes.
+* Mixing two unrelated functional changes in the same commit.
+* Sending large new features in a single commit.
+
+Expected git commit message structure:
+
+* The first line should be limited to 50 characters, should not end with a full stop and the first word should be capitalised.
+* Use the imperative mood in the subject line - e.g. ```Fix a typo in CONTRIBUTING.md```, ```Remove if/else block in myfile.sh```, ```Eat your dinner```.
+* Insert a single blank line after the first line.
+* Subsequent lines should be wrapped at 72 characters.
+* Provide a detailed description of the change in the following lines, using the guidelines in the section below.
+* The last line of the commit should be a reference to the issue being fixed using the JIRA issue Key e.g. ```RE-1500```
```

Comment:
Highlighting this because it's different from what we've been doing, generally we've been adding this as the first word of the commit message title. I'd prefer us to start adding it at the end of the message because having it in the title takes up precious characters.

Reply:
Should we prefix the card number with something? Otherwise it's kinda weird having this weird thing hanging about which will mean nothing if we move to another system. eg: ``JIRA: RE-1500``